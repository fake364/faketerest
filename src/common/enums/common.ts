export enum GENDER_ENUM {
  FEMALE = -1,
  NOT_SPECIFIED = 0,
  MALE
}
