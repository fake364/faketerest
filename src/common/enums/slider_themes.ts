export enum SLIDER_THEMES {
	DINNER_IDEA = 'dinnerIdea',
	HOME_DECOR = 'homeDecor',
	NEW_OUTFIT = 'newOutfit',
	THUMB_IDEA = 'thumbIdea'
}
