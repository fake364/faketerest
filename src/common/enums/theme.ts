export enum THEME_TYPE {
	BASE = 'base',
	BUSINESS = 'business',
	NEWSROOM = 'newsroom'
}