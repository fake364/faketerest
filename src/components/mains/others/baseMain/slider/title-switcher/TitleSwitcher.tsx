import React from 'react';
import { useTranslation } from 'next-i18next';
import { SLIDER_THEMES } from '../../../../../../common/enums/slider_themes';

type Props = { currentTheme: SLIDER_THEMES };

const TitleSwitcher: React.FC<Props> = ({ currentTheme }) => {
	const { t } = useTranslation('common');

	return (
		<div className="flex-1 flex flex-col items-center mt-24 relative">
			<span className="text-[60px] font-normal">
				{t('titles.base.mainPage.slider.getYourNext')}
			</span>
			<span
				slider-theme={currentTheme}
				className="text-slider_theme text-[50px] animate-move_and_disappear ease-out"
			>
				{t(`titles.base.mainPage.slider.${currentTheme}`)}
			</span>
		</div>
	);
};

export default TitleSwitcher;
