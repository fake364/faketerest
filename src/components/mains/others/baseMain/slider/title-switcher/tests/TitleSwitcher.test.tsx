import React from 'react';
import { shallow } from 'enzyme';
import TitleSwitcher from '../TitleSwitcher';
import { SLIDER_THEMES } from '../../../../../../../common/enums/slider_themes';

describe('TitleSwitcher', () => {
  it('should match snapshot when is not animated', () => {
    const component = shallow(
      <TitleSwitcher isAnimation={false} shownType={SLIDER_THEMES.NEW_OUTFIT} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should add additional styles when it is animated', () => {
    const component = shallow(
      <TitleSwitcher isAnimation={true} shownType={SLIDER_THEMES.NEW_OUTFIT} />
    );
    expect(component).toMatchSnapshot();
  });
});
