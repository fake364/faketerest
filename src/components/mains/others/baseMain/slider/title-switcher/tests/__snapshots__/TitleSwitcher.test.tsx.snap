// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleSwitcher should add additional styles when it is animated 1`] = `
<div
  className="flex-1 flex flex-col items-center mt-24 relative"
>
  <span
    className="text-[60px] font-normal"
  >
    titles.base.mainPage.slider.getYourNext
  </span>
  <span
    className="text-slider_theme text-[50px] animate-move_and_disappear ease-out"
    slider-theme="newOutfit"
  >
    titles.base.mainPage.slider.newOutfit
  </span>
</div>
`;

exports[`TitleSwitcher should match snapshot when is not animated 1`] = `
<div
  className="flex-1 flex flex-col items-center mt-24 relative"
>
  <span
    className="text-[60px] font-normal"
  >
    titles.base.mainPage.slider.getYourNext
  </span>
  <span
    className="text-slider_theme text-[50px]"
    slider-theme="newOutfit"
  >
    titles.base.mainPage.slider.newOutfit
  </span>
</div>
`;
